flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/ExecutionGraphCacheTest.java

```diff
@@ -199,12 +212,17 @@ public void testCacheEntryCleanup() throws Exception {
 	public void testConcurrentAccess() throws Exception {
 		final Time timeout = Time.milliseconds(100L);
 		final Time timeToLive = Time.hours(1L);
-		final JobID jobId = new JobID();
-
-		final AccessExecutionGraph accessExecutionGraph = mock(AccessExecutionGraph.class);
 
-		final JobManagerGateway jobManagerGateway = mock(JobManagerGateway.class);
-		when(jobManagerGateway.requestJob(eq(jobId), any(Time.class))).thenReturn(CompletableFuture.completedFuture(accessExecutionGraph));
+		final AtomicInteger requestJobCalls = new AtomicInteger(0);
```

Comment:
Wouldn't it be easier to make `CountingRestfulGateway` thread safe, and reuse it here?

Reply:
True, will change it.